api/src/pcapi/core/token/__init__.py

```diff
@@ -73,8 +86,14 @@ def load_and_check(cls, encoded_token: str, type_: TokenType, user_id: int | Non
     def create(cls, type_: TokenType, ttl: timedelta | None, user_id: int, data: dict | None = None) -> "Token":
         encoded_token = utils.encode_jwt_payload({"token_type": type_.value, "user_id": user_id, "data": data})
         app.redis_client.set(cls._get_redis_key(type_, user_id), encoded_token, ex=ttl)  # type: ignore [attr-defined]
-        return Token.load_without_checking(encoded_token)
+        token = Token.load_without_checking(encoded_token)
+        cls._log(cls._TokenAction.CREATE, token)
+        return token
 
     @classmethod
     def token_exists(cls, type_: TokenType, user_id: int) -> bool:
         return app.redis_client.exists(cls._get_redis_key(type_, user_id))  # type: ignore [attr-defined]
+
+    @classmethod
+    def _log(cls, action: _TokenAction, token: "Token") -> None:
+        logger.info("[TOKEN](%s){%i, %s, %s}", action, token.user_id, token.type_.value, token.encoded_token)
```

Comment:
Puisque tu as transformé la fonction en méthode, tu peux éviter d'avoir à passer le token. (Pour cela, il ne faut pas que ce soit une "classmethod", mais je ne pense pas qu'on en ait besoin.)

```suggestion
    def _log(self, action: _TokenAction) -> None:
        logger.info("[TOKEN](%s){%i, %s, %s}", action, self.user_id, self.type_.value, self.encoded_token)
```

Reply:
je l'ai fait parceque create est @classmethod et ça utilise mais je peux contourner